### PR TITLE
Generator remove duplicates

### PR DESCRIPTION
**Pull request by jonathanadler**
I'll write more later

This branch is forked from Shlomi's branch: https://github.com/wix/wix-ui-tpa/pull/463

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_467/

File: docs/USAGE.md [hunk 1]
```diff
@@ -68,14 +59,51 @@ The component's style can be customized like this:
...
+
+#### An optimized way for style overriding - New!
+Starting of version 2.10.0, new components will support a more optimized way to override styles, which **reduces bundle size significantly**.
+
+*Note*: This will become the official way to override styles, however not all components are compatible yet (please check each component's documentation page to see if it supports the new way).
+
+```css
+/* CheckoutButton.st.css */
+:import {
+    -st-from: "wix-ui-tpa/dist/src/components/Button/Button.st.css";
```
Comment (sivanharel) [inline docs/USAGE.md:71]: I think we should give an example of overriding a component that has the 'overrideStyleParams'.
Maybe we should change this example to theme the Badge component? or to implement the overrideStyleParams in the button component.  WDYT?

File: .wuf/generator/templates/Component/Component.spec.tsx [hunk 2]
```diff
@@ -3,27 +3,28 @@ import { createUniDriverFactory } from 'wix-ui-test-utils/uni-driver-factory';
...
+    return createDriver(
+      // <TPAComponentsProvider value={contextProps}>
+        <{%ComponentName%} {...props} />
+      // </TPAComponentsProvider>
+    );
+  };
+
   it('should render', async () => {
-    const driver = createDriver(<{%ComponentName%} buttonText="Click Me" />);
+    const driver = bootstrap();
```
Comment (sivanharel) [inline .wuf/generator/templates/Component/Component.spec.tsx:24]: Why did you change it for that?

File: .wuf/generator/templates/Component/Component.e2e.ts [hunk 3]
```diff
@@ -20,8 +17,7 @@ describe('{%componentName%}', () => {
   beforeEach(() => browser.get(storyUrl));
 
   it('should render', async () => {
-    const driver = {%componentName%}TestkitFactory({ dataHook });
-    await waitForVisibilityOf(await driver.element(), 'Cannot find {%ComponentName%}');
-    expect((await driver.element()).isDisplayed()).toBe(true);
+    // If the component needs a real browser environment for testing, use this file
+    // If not, please remove this file
```
Comment (sivanharel) [inline .wuf/generator/templates/Component/Component.e2e.ts:21]: Maybe you should keep the code as well. I think it's a good reference.

**Review by sivanharel — COMMENTED** [review]
Looks very good!
All the comments that you added for the code and for the documentation doc are very informative :)
I added some comments, let me know what do you think.

File: docs/USAGE.md (hunk 1 above)
Comment (shlomitc) [reply docs/USAGE.md:71]: sounds like a good idea

Comment (jonathanadler) [reply docs/USAGE.md:71]: Yeah 👍

File: .wuf/generator/templates/Component/Component.spec.tsx (hunk 2 above)
Comment (jonathanadler) [reply .wuf/generator/templates/Component/Component.spec.tsx:24]: It's just to give a generic method to be used in all tests.
Instead of copy-pasting `createDriver(<MyComponent />)` in each test.
Especially if it's wrapped with the `TPAComponentsProvider`.

Btw, I see that I forgot to pass props to the `bootstrap` method.
I'll fix that.

File: .wuf/generator/templates/Component/Component.e2e.ts (hunk 3 above)
Comment (jonathanadler) [reply .wuf/generator/templates/Component/Component.e2e.ts:21]: I was concerned that if there'll be some code here, it will discourage developers to delete the file 😄

File: .wuf/generator/templates/Component/Component.e2e.ts [hunk 4]
```diff
@@ -1,15 +1,12 @@
 import { browser } from 'protractor';
```
[the remark this answers is not in the document]
Comment (jonathanadler) [reply .wuf/generator/templates/Component/Component.e2e.ts:1]: I was very close to removing this file, and I agree with the points you mentioned.
It's just that adding an e2e test can be a bit complicated.
So I thought leaving this file can make it easier.

But yeah, I think I'll remove the file.
I'll add a document to explain how to add e2e tests in case one's needed.

[the remark this answers is not in the document]
Comment (shlomitc) [reply .wuf/generator/templates/Component/Component.e2e.ts]: we'll do it in a separate PR

**Review by shlomitc — APPROVED** [review]
